packages/use-intl/src/core/createFormatter.tsx

```diff
@@ -284,7 +284,7 @@ export default function createFormatter({
   type FormattableListValue = string | ReactElement;
   function list<Value extends FormattableListValue>(
     value: Iterable<Value>,
-    formatOrOptions?: string | Intl.ListFormatOptions
+    formatOrOptions?: IntlFormats['list'] | Intl.ListFormatOptions
```

Comment:
I noticed for `format.list` this doesn't work yet, I guess this is why:

```suggestion
    formatOrOptions?: keyof IntlFormats['list'] | Intl.ListFormatOptions
```

- [ ] Fix type checking for `.list`